docs/user/exercises/programming-exercise-AI-proof-guidelines.inc

```diff
@@ -0,0 +1,491 @@
+
+-  **Text in picture format:** Giving the text exercise instructions in
+   a picture format that can not be copied. Rather than writing long
+   inquiries into the AI, the students will be promoted to asking
+   reasonable questions and solve exercises on their own.
+
+   .. note::
+      Since there are already possibilities to copy text from pictures, the effect of
+      this measure are limited. More suitable for exam mode.
+
+-  **Java frameworks:** Using Java frameworks, such as JavaFX, is challenging for AI
+   and helps to visualize projects in a more interactive way
+   for students through gamification. How can we incorporate that for
+   our exercises?
+
+   -  Provide Graphical User Interfaces (GUI) for
+      students with which they can also test the functionality of their
+      code.
+
+   -  Give exercises that use GUIs because AI has difficulty
+      solving them accurately.
+
+   -  Provide pictures of how the GUI should look in the end.
+
+      Students needed to implement the GUI based on the this figure that
+      was given in the problem statement.
+
+      .. figure:: programming/AI-proof-exercises/GUI-upgrade-for-game.png
+         :align: center
+         :width: 30%
+
+
+         Example GUI from homework exercise
+
+   .. note::
+      Apply this approach when students already have knowledge of what a GUI is. Most likely in
+      a later stage of the semester.
+
+.. note::
+   ChatGPT is limited to text input. Including a fair amount of graphical
+   representation for the exercises helps to limit the effectiveness of such tools.
+
+Problem statement and exercise instructions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+
+-  **Exercise length:** There is a negative correlation between the
+   length of exercises and AI performance:
+
+   -  Do not keep the exercises between 500 and 1000 characters, but
+      rather keep them between 1200 and 2500 characters. Short exercises
+      are normally easier to solve for AI since there are not so many
+      constraints and the problem statement only focused on the most
+      relevant implementation parts.
+
+
+-  **Uncommon data types:** Using types that aren't commonly used in Java
+   programming can confuse the AI.
+   Most times, it tries to solve the instructions by finding a way
+   around using these types and only later converts them without using
+   the appropriate methods that including these data types. Complex data
+   types, such as custom classes and data structures, often involve
+   additional levels of complexity and variation in their
+   implementation.
+
+   Here are some of the complex data types that can improve the AI
+   security of the exercises:
+
+   -  Duration
+
+   -  BigInteger, BigDecimal
+
+   -  LocalDate
+
+   -  LocalTime
+
+   -  Locale
+
+   -  StringBuilder
+
+   -  Currency
+
+   -  Custom classes implemented in the template repository
```

Comment:
From my experience ChatGPT can easily handle:
- BingInteger, BigDecimal
- StringBuilder
- Custom classes as long as you provide them
- Date and Time stuff (though this may vary between different implementations)